x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -37,6 +37,17 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 		return errorLogAndFormat("data_reliability_claim", map[string]string{"error": err.Error()}, "error creating dataReliabilityByConsumer")
 	}
 
+	addressEpochBadgeMap := map[string]types.Badge{}
+	for _, relay := range msg.Relays {
+		if relay.Badge != nil {
+			mapKey := relay.Badge.Address + "_" + strconv.FormatUint(relay.Badge.Epoch, 10)
+			_, ok := addressEpochBadgeMap[mapKey]
+			if !ok {
```

Comment:
if not ok return error, same address used 2 different badges

Reply:
done